src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs

```diff
@@ -129,55 +129,32 @@ internal TracerProviderBuilder AddProcessor(BaseProcessor<Activity> processor)
             return this;
         }
 
-        internal TracerProvider Build()
-        {
-            return new TracerProviderSdk(
-                this.resourceBuilder.Build(),
-                this.sources,
-                this.diagnosticSourceInstrumentationFactories,
-                this.instrumentationFactories,
-                this.sampler,
-                this.processors);
-        }
-
         /// <summary>
-        /// Adds a DiagnosticSource based instrumentation.
-        /// This is required for libraries which is already instrumented with
-        /// DiagnosticSource and Activity, without using ActivitySource.
+        /// Adds activity with a given operation name to the list of subscribed activities. This is only for legacy activities (i.e. activities created without using ActivitySource API).
         /// </summary>
-        /// <typeparam name="TInstrumentation">Type of instrumentation class.</typeparam>
-        /// <param name="instrumentationFactory">Function that builds instrumentation.</param>
+        /// <param name="operationName">OperationName to add.</param>
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
-        internal TracerProviderBuilder AddDiagnosticSourceInstrumentation<TInstrumentation>(
-            Func<ActivitySourceAdapter, TInstrumentation> instrumentationFactory)
-            where TInstrumentation : class
+        internal TracerProviderBuilder AddLegacyActivity(string operationName)
```

Comment:
nit: we could change this to `params string[]`, we have many cases calling this twice.